services/src/main/scala/cromwell/services/cost/CostCatalogService.scala

```diff
@@ -0,0 +1,110 @@
+package cromwell.services.cost
+
+import akka.actor.{Actor, ActorRef}
+import com.google.cloud.billing.v1._
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import common.util.StringUtil.EnhancedToStringable
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+
+import java.io.{FileOutputStream, ObjectOutputStream}
+import java.time.{Duration, LocalDateTime}
+import scala.jdk.CollectionConverters.IterableHasAsScala
+import java.time.temporal.ChronoUnit.HOURS
+case class CostCatalogMessage() extends ServiceRegistryMessage {
+  override val serviceName = "CostCatalogService"
+}
+
+case class ExpiringCostCatalog(catalog: Map[CostCatalogKey, CostCatalogValue], fetchTime: LocalDateTime)
+
+/**
+ * This actor handles the fetching and caching of the public Google Cost Catalog.
+ * This enables us to look up the SKUs necessary for cost calculations.
+ */
+class CostCatalogService(serviceConfig: Config,
+                         globalConfig: Config,
+                         serviceRegistry: ActorRef,
+                         mockTestData: Option[List[Sku]] = Option.empty,
+                         maxCatalogLifetime: Duration = Duration.of(24, HOURS)
+) extends Actor
+    with LazyLogging {
+
+  // Cached catalog. Refreshed lazily when older than maxCatalogLifetime.
+  private var costCatalog: ExpiringCostCatalog =
+    ExpiringCostCatalog(CostCatalogService.processCostCatalog(fetchNewCatalog), java.time.LocalDateTime.now())
```

Comment:
I've forgotten most of what I know about Java datetimes, but I think we typically avoid using local ones for things like this.